server/service/clustered_storage.go

```diff
@@ -34,6 +34,7 @@ func (d *clusterStorageDriverInstance) GetProjectStorage(project string) storage
 	}
 }
 
+// This is used to prepare the storage driver for pushing objects
```

Comment:
Not really. `clusterStorageProjectDriverInstance` is more a handle to the storage driver that is specific for the current project. It's a way to tell the storage driver "I now want to talk about objects in this particular project", which ensures that there's a strict separation between objects in different projects.
As you can see in the general storage tests, there's a very strict requirement that unless an object is written for project `A` *and* `B`, requesting an object from a `clusterStorageProjectDriverInstance` scoped to project `A`, should *not* return objects from project `B`.